src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -5524,7 +5524,12 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
               context.pendingStabilizers ::= vdef
               qual.changeOwner(context.owner -> vsym)
               val newQual = Ident(vsym) setType singleType(NoPrefix, vsym) setPos qual.pos.focus
-              return typedSelect(tree, newQual, name)
+              return typedSelect(tree, newQual, name).modifyType(_.map {
+                // very specific fix for scala/bug#12987 (details in the ticket)
+                case t: AliasTypeRef if t.pre.termSymbol == vsym && context.undetparams.contains(t.normalize.typeSymbol) =>
+                  t.normalize
+                case t => t
+              })
```

Comment:
In short, given a method `def m(b: FT, o: Option[I]): Int` and a stabilizer is inserted at some call site `stabilizer$1.m(b, None)`

If `FT` is a type parameter, `stabilizer$1.m` has a MethodType where the parameter type `FT` is subsituted by asSeenFrom.

If `FT` is a type member, the parameter type ends up being `AliasNoArgsTypeRef(stabilizer$1.type, FT)`, which is equivalent in theory, but `instantiateTypeParams` doesn't see through this alias.

More details in https://github.com/scala/bug/issues/12987.